### DISCUSSION TITLE
How can I solve a bug caused by neogit together with lazyvim?

### DISCUSSION THREAD

**Discussion by r-darwish** [LazyVim/LazyVim#4616] [Q&A] [answered]
I'm using Neogit together with Lazyvim and noticed that hitting `C-c C-c` in the commit window while in insert mode causes havoc, so I opened [this discussion](https://github.com/NeogitOrg/neogit/discussions/1492#discussioncomment-10801681). The maintainer suggested that this is cause by nvim-cmp and suggested a solution. Based on this solution [I made a PR](https://github.com/hrsh7th/nvim-cmp/pull/2053) which seems to fix it. However, it has not been merged yet and I rather not keep using a fork until it does. I tried setting `nvim-cmp` back to the upstream remote and tried solving this issue with autocmd:

```lua
vim.api.nvim_create_autocmd({
  "BufWinLeave",
}, {
  callback = function()
    local cmp = require("cmp")
    cmp.close()
  end,
})
```
However, it didn't work and this autocmd is called way more times than I've expected. Is there any hack I can apply to my own configuration to mitigate this issue until my PR is merged?

**Comment by dpetka2001** [comment, marked as answer]
From `:h i_CTRL-C` and this [comment by a Neovim core member](https://github.com/neovim/neovim/pull/29581#issuecomment-2318966355), it seems to me that hitting `<C-c>` in `insert` mode and has some side-effects. The dev says this is intentional and users should be wary of it. So, just don't use `<C-c>`? I haven't dug deep into it, but I believe your best choice is to use your forked repo until it gets merged. It looks to me that pressing `<C-c>` in `insert` mode is something like a hard exit from to `normal` mode.
